### PR TITLE
Move some stuff to modules

### PR DESCRIPTION
**Pull request by RedSparr0w**
This PR just moves some more stuff to modules.

### CODE REVIEW

File: src/modules/gems/GemDeal.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        '-1': undefined,
+        0: undefined,
+        1: undefined,
+        2: undefined,
+        3: undefined,
+        4: undefined,
+        5: undefined,
+        6: undefined,
+        7: undefined,
+        8: undefined,
```
Comment (Aegyo) [inline src/modules/gems/GemDeal.ts:30]:
```suggestion
        [Region.none]: undefined,
        [Region.kanto]: undefined,
        [Region.johto]: undefined,
        [Region.hoenn]: undefined,
        [Region.sinnoh]: undefined,
        [Region.unova]: undefined,
        [Region.kalos]: undefined,
        [Region.alola]: undefined,
        [Region.galar]: undefined,
        [Region.final]: undefined,
```

File: src/modules/items/EggItem.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }
+            default: {
+                const etype = EggType[EggItemType[this.type].split('_')[0]];
+                return App.game.breeding.getTypeCaughtStatus(etype);
+            }
+        }
+    }
+}
+
+ItemList.Fire_egg = new EggItem(EggItemType.Fire_egg, 1000, undefined, 'Fire Egg');
```
Comment (Aegyo) [inline src/modules/items/EggItem.ts:54]: I've been doing all of these in ItemList.ts, so that anything that imports EggItem doesn't also depend on ItemList

File: src/modules/items/FluteItem.ts [hunk 3]
```diff
@@ -63,12 +71,11 @@ class FluteItem extends Item {
...
-
 }
 
-ItemList.Yellow_Flute       = new FluteItem(GameConstants.FluteItemType.Yellow_Flute, 'Pokémon Attack', ['Grass', 'Flying', 'Electric'], 'pokemonAttack', 1.02);
-ItemList.Time_Flute        = new FluteItem(GameConstants.FluteItemType.Time_Flute, 'Gym and Dungeon Timers', ['Ground', 'Poison', 'Steel'], undefined, 1.02);
-ItemList.Black_Flute        = new FluteItem(GameConstants.FluteItemType.Black_Flute, 'Click Attack', ['Dark', 'Psychic', 'Fighting'], 'clickAttack', 1.02);
-ItemList.Red_Flute         = new FluteItem(GameConstants.FluteItemType.Red_Flute, 'Egg Steps', ['Fire', 'Rock', 'Dragon'], 'eggStep', 1.02);
-ItemList.White_Flute         = new FluteItem(GameConstants.FluteItemType.White_Flute, 'Shiny Chance', ['Normal', 'Fairy', 'Ice'], 'shiny', 1.02);
-ItemList.Blue_Flute        = new FluteItem(GameConstants.FluteItemType.Blue_Flute, 'EV Yield', ['Water', 'Bug', 'Ghost'], 'ev', 1.02);
+ItemList.Yellow_Flute = new FluteItem(FluteItemType.Yellow_Flute, 'Pokémon Attack', ['Grass', 'Flying', 'Electric'], 'pokemonAttack', 1.02);
```
Comment (Aegyo) [inline src/modules/items/FluteItem.ts:76]: Same comment as with EggItem

File: src/modules/items/Vitamin.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    use(): boolean {
+        return true;
+    }
+
+    get image() {
+        return `assets/images/items/vitamin/${this.displayName}.png`;
+    }
+}
+
+ItemList.RareCandy = new Vitamin(VitaminType.RareCandy, Infinity, undefined, undefined, 'Rare Candy', 'A rare-to-find candy that currently has no use.');
```
Comment (Aegyo) [inline src/modules/items/Vitamin.ts:24]: Same comment as with EggItem

File: src/modules/temporaryWindowInjection.ts [hunk 5]
```diff
@@ -310,9 +318,13 @@ Object.assign(<any>window, {
...
     CaughtIndicatingItem,
+    EggItem,
+    FluteItem,
+    MegaStoneItem,
     PokeballItem,
+    Vitamin,
+    VitaminController,
     RoamingPokemonList,
     DataPokemon,
-    MegaStoneItem,
```
Comment (Aegyo) [inline src/modules/temporaryWindowInjection.ts:316]: Why was this removed? I think it is needed in DungeonRunner to check if an item is an instanceof MegaStoneItem

Comment (RedSparr0w) [reply src/modules/temporaryWindowInjection.ts:316]: Was moved up a bit, to be with the other items.

File: src/modules/items/EggItem.ts (hunk 2 above)
Comment (RedSparr0w) [reply src/modules/items/EggItem.ts:54]: Ah, yeah makes sense, will move em all over 👍

File: src/modules/items/FluteItem.ts (hunk 3 above)
Comment (RedSparr0w) [reply src/modules/items/FluteItem.ts]: The flutes seem to have quite the dependency cycle when they move to `ItemList.ts`.
Gonna have to leave them out of this PR.

**Review by Jaaslet — APPROVED** [review]
Looks good. Thanks for the PR
